pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -232,6 +248,23 @@ def newpkgcb(pkgId, name, arch):
             result = await downloader.run()
             metadata_pb.increment()
 
+            kickstart_parsed = {}
+            namspaces = [".treeinfo", "treeinfo"]
+            for namespace in namspaces:
+                response = requests.get(urljoin(self.remote.url, namespace))
+                if response.status_code == 200:
+                    downloader = self.remote.get_downloader(
+                        url=urljoin(self.remote.url, namespace)
+                    )
+                    kickstart_result = await downloader.run()
+                    kickstart_path = kickstart_result.path
+                    kickstart = TreeInfo()
+                    kickstart.load(f=kickstart_path)
+                    parser = SortedConfigParser()
+                    kickstart.serialize(parser)
+                    kickstart_parsed.update(parser._sections)
```

Comment:
you could probably do away with kickstart_parsed and instead build and put() the KickstartTree units directly here before you `break`.